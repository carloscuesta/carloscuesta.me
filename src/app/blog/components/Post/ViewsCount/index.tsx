'use client'

import useViewsCount from 'src/utils/hooks/useViewsCount'

type Props = { slug: string; views?: string }

const ViewsCount = ({ slug, views }: Props) => {
  const { data } = useViewsCount(slug, views)

  return <>{data?.views?.[slug]}</>
}

export default ViewsCount
