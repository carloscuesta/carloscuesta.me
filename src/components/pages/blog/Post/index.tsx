import Link from 'next/link'

import { type PostPreview } from 'src/utils/api/blog/mutators'

const Post = ({ post }: { post: PostPreview }) => (
  <li key={post.slug} className='rounded-lg font-semibold transition-all hover:bg-neutral-100 focus-within:bg-neutral-100 dark:hover:bg-neutral-900 dark:focus-within:bg-neutral-900 -mx-3'>
    <Link
      href={`/blog/${post.slug}`}
      title={post.title}
      className='rounded-lg flex flex-col-reverse gap-1 sm:flex-row sm:gap-3 px-3 py-3 sm:py-4 outline-none focus:ring-2 focus:ring-neutral-300 dark:focus:ring-neutral-500'
    > 
      <time
        className='opacity-60 sm:w-20'
        dateTime={post.datePublished.value}
        aria-label={`Written ${post.datePublished.formatDate}`}
      >
        {post.datePublished.formatMonthDay}
      </time>

      <p>{post.title}</p>
    </Link>
  </li>
)

export default Post
