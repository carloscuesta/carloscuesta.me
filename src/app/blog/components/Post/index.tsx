import Link from 'next/link'

import { type PostPreview } from 'src/utils/api/blog/mutators'
import ViewsCount from 'src/components/ViewsCount'

const Post = ({ post, views }: { post: PostPreview; views: string }) => (
  <li
    key={post.slug}
    className="-mx-3 rounded-lg transition-all focus-within:bg-neutral-100 hover:bg-neutral-100 dark:focus-within:bg-neutral-900 dark:hover:bg-neutral-900"
  >
    <Link
      href={`/blog/${post.slug}`}
      title={post.title}
      className="flex flex-col-reverse gap-1 rounded-lg px-3 py-3 outline-none focus:ring-2 focus:ring-neutral-300 dark:focus:ring-neutral-500 sm:flex-row sm:gap-3 sm:py-4"
    >
      <time
        className="opacity-60 sm:w-20"
        dateTime={post.datePublished.value}
        aria-label={`Written ${post.datePublished.formatDate}`}
      >
        {post.datePublished.formatMonthDay}
      </time>

      <div className="grid flex-1 grid-flow-col items-baseline justify-between gap-10 md:items-center">
        <p className="font-medium">{post.title}</p>
        <p className="font-mono text-sm opacity-60">
          <ViewsCount slug={post.slug} views={views} />
        </p>
      </div>
    </Link>
  </li>
)

export default Post
