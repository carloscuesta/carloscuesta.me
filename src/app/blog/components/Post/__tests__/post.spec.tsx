import renderer from 'react-test-renderer'

import Post from '../'
import * as stubs from './stubs'

describe('Post', () => {
  it('should render the component', () => {
    const wrapper = renderer.create(<Post post={stubs.post} />)

    expect(wrapper).toMatchSnapshot()
  })
})
