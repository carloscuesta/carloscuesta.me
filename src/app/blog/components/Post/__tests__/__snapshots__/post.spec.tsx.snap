// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post should render the component 1`] = `
<li
  className="-mx-3 rounded-lg transition-all focus-within:bg-neutral-100 hover:bg-neutral-100 dark:focus-within:bg-neutral-900 dark:hover:bg-neutral-900"
>
  <a
    className="flex flex-col-reverse gap-1 rounded-lg px-3 py-3 outline-none focus:ring-2 focus:ring-neutral-300 dark:focus:ring-neutral-500 sm:flex-row sm:gap-3 sm:py-4"
    href="/blog/post-blog-slug"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
    title="A post"
  >
    <time
      aria-label="Written 15/12/2019"
      className="opacity-60 sm:w-20"
      dateTime="2019-12-15 10:00"
    >
      15 Dec
    </time>
    <p
      className="font-medium"
    >
      A post
    </p>
  </a>
</li>
`;
